larpandoracontent/LArHelpers/LArMCParticleHelper.cc

```diff
@@ -294,33 +430,58 @@ void LArMCParticleHelper::SelectReconstructableMCParticles(const MCParticleList
 
     // Obtain vector: primary mc particles
     MCParticleVector mcPrimaryVector;
-    LArMCParticleHelper::GetPrimaryMCParticleList(pMCParticleList, mcPrimaryVector);
+
+    if (!testBeamHierarchyMode)
+    {
+        LArMCParticleHelper::GetPrimaryMCParticleList(pMCParticleList, mcPrimaryVector);
+    }
+    else
+    {
+        LArMCParticleHelper::GetLeadingMCParticleList(pMCParticleList, mcPrimaryVector);
+    }
 
     // Select MCParticles matching criteria
     MCParticleVector candidateTargets;
     LArMCParticleHelper::SelectParticlesMatchingCriteria(mcPrimaryVector, fCriteria, candidateTargets);
 
     // Ensure the MCParticles have enough "good" hits to be reconstructed
     LArMCParticleHelper::SelectParticlesByHitCount(candidateTargets, mcToTrueHitListMap, mcToPrimaryMCMap, parameters, selectedMCParticlesToHitsMap);
+
+    // ATTN: The parent particle must be in the hierarchy map, event if not reconstructable
+    if (testBeamHierarchyMode)
```

Comment:
Was this not required before the introduction of the test beam hierarchy mode? (I'm here just looking for ways of sharing implementation between the with and without test beam hierarchy modes). Currently leaning towards two separate functions, each maybe then calling a block of common implementation, rather than one function with a Boolean argument/switch.